indico/modules/categories/templates/display/sidebar.html

```diff
@@ -1,89 +1,179 @@
+                        </span>
+                    {% endif %}
+                </span>
+            </a>
+        </ind-attachment>
+    </li>
+{% endmacro %}
+
+{% macro _render_attachment_folder(attachment) %}
+    {% set folder_name = attachment.title %}
+    <li>
+        <details open>
+            <summary aria-label="folder {{ attachment.title }}">
+                {{ attachment.title }}
+            </summary>
+            <ul>
+                {% for child in attachment.attachments %}
+                    {{ _render_attachment_file(child) }}
+                {% endfor %}
+            </ul>
+        </details>
+    </li>
+{% endmacro %}
 
 {% macro render_attachments(category) %}
     {%- set can_manage = category.can_manage(session.user) -%}
     {%- if category.attachment_count or can_manage %}
-        <div class="title">
-            <h2 class="icon-attachment">{% trans %}Materials{% endtrans %}</h2>
-            {% if can_manage %}
-                <div class="actions">
-                    <button class="i-button text-color subtle icon-edit js-manage-attachments"
-                            title="{% trans %}Manage materials{% endtrans %}"
-                            data-title="{% trans title=category.title %}Manage category materials{% endtrans %}"
-                            data-locator="{{ category.locator|tojson|forceescape }}"
-                            data-attachment-editor
-                            data-reload-on-change></button>
+        <section id="material-list" aria-labelledby="materials">
+            <header>
+                <h2 id="materials">{% trans %}Materials{% endtrans %}</h2>
+                {% if can_manage %}
+                    <div class="actions">
+                        <button class="i-button text-color subtle icon-edit js-manage-attachments"
+                                title="{% trans %}Manage materials{% endtrans %}"
+                                data-title="{% trans title=category.title %}Manage category materials{% endtrans %}"
+                                data-locator="{{ category.locator|tojson|forceescape }}"
+                                data-attachment-editor
+                                data-reload-on-change></button>
+                    </div>
+                {% endif %}
+            </header>
+            <ul>
+                {% set attachments = category.attached_items %}
+                {% if attachments %}
+                    {% for attachment in attachments.files %} {{ _render_attachment_file(attachment) }}{% endfor %}
+                    {% for attachment in attachments.folders %}{{ _render_attachment_folder(attachment) }}{% endfor %}
+                {% else %}
+                    <p class="empty-list-notice">{% trans %}There are no materials yet.{% endtrans %}</p>
+                {% endif %}
+            </ul>
+        </section>
+
+        <ind-attachment-preview-modal>
```

Comment:
I see you're just adding this on the category page. Are you also replacing the previewer for event-level attachments? If yes, you may want to move this into a separate template to avoid repeating the same thing in two places.

Reply:
> Are you also replacing the previewer for event-level attachments?

Yeah, maybe later, when I come around to that.

Reply:
I'll extract it then.